runtime/common/src/migrations/liquidity_pools_v2.rs

```diff
@@ -0,0 +1,846 @@
+					);
+
+					writes.saturating_accrue(2);
+				} else {
+					writes.saturating_accrue(1);
+					continue;
+				}
+			}
+		}
+
+		fn migrate_failed_message_queue(mut reads: u64, mut writes: u64) {
+			let n: u64 = FailedOutboundMessages::<T>::iter_keys()
+				.count()
+				.saturated_into();
+			log::info!("{LOG_PREFIX}: Initiating migration of {n} failed outbound messages");
+			reads.saturating_accrue(n);
+
+			for (nonce, (domain, _, message, err)) in FailedOutboundMessages::<T>::iter().drain() {
+				if let Some(router_id) = maybe_router_id(domain) {
+					log::info!(
+						"{LOG_PREFIX}: Migrating failed outbound domain {domain:?} message {message:?}"
+					);
+
+					// Should never be none since target are outbound messages
+					pallet_liquidity_pools_gateway_queue::FailedMessageQueue::<T>::insert(
+						nonce,
+						(GatewayMessage::Outbound { message, router_id }, err),
+					);
+					writes.saturating_accrue(2);
+				} else {
+					writes.saturating_accrue(1);
+					continue;
+				}
+			}
+		}
+	}
+}
+
+pub mod init_axelar_router {
+	use cfg_types::EVMChainId;
+	#[cfg(feature = "try-runtime")]
+	use frame_support::storage::transactional;
+	use frame_support::{
+		dispatch::DispatchResult,
+		traits::{GetStorageVersion, OriginTrait, StorageVersion},
+	};
+	use frame_system::pallet_prelude::OriginFor;
+	#[cfg(feature = "try-runtime")]
+	use sp_arithmetic::traits::SaturatedConversion;
+	use sp_arithmetic::traits::Saturating;
+	use sp_std::boxed::Box;
+
+	use super::{
+		types::{v0, v2},
+		*,
+	};
+	pub struct Migration<T>(PhantomData<T>);
+
+	const LOG_PREFIX: &str = "DomainRoutersToAxelarConfig";
+
+	impl<T> OnRuntimeUpgrade for Migration<T>
+	where
+		T: pallet_liquidity_pools_gateway::Config
+			+ pallet_liquidity_pools_gateway_queue::Config
+			+ pallet_xcm_transactor::Config
+			+ pallet_ethereum_transaction::Config
+			+ pallet_axelar_router::Config
+			+ pallet_evm::Config
+			+ frame_system::Config,
+		T::AccountId: AsRef<[u8; 32]>,
+		OriginFor<T>:
+			From<pallet_ethereum::Origin> + Into<Result<pallet_ethereum::Origin, OriginFor<T>>>,
+		v0::DomainRouter<T>: sp_std::fmt::Debug,
+		v2::DomainRouter<T>: sp_std::fmt::Debug,
+	{
+		fn on_runtime_upgrade() -> Weight {
+			let (reads, writes) = Self::migrate_domain_routers().unwrap_or_default();
+			log::info!(
+				"{LOG_PREFIX} ON_RUNTIME_UPGRADE: Migration done with {writes:?} updated domains!"
+			);
+
+			T::DbWeight::get().reads_writes(reads, writes.saturating_add(1))
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			let mut writes = 0u64;
+
+			// Need to rollback in order to be NOOP
+			let _ = transactional::with_storage_layer(|| -> DispatchResult {
+				let (r, w) = Self::migrate_domain_routers()?;
+				log::info!("{LOG_PREFIX} PRE Migration counts {w:?} updated domains and {} removed domains", r.saturating_sub(w));
+				writes = w;
+				Err(DispatchError::Other("Reverting on purpose"))
+			});
```

Comment:
Note: There are two `pre_upgrade` hooks in this PR which utilize a transactional layer to ensure the exact migration call which is applied in the corresponding `on_runtime_upgrade` does not fail. However, since those migration calls mutate storage, I am reverting the storage changes by failing the transactional layer on purpose.